numpy/random/mtrand/randomkit.c

```diff
@@ -240,6 +246,101 @@ rk_random(rk_state *state)
     return y;
 }
 
+
+/*
+ * Returns an unsigned 64 bit random integer.
+ */
+NPY_INLINE static npy_uint64
+rk_uint64(rk_state *state)
+{
+    npy_uint64 upper = (npy_uint64)rk_random(state) << 32;
+    npy_uint64 lower = (npy_uint64)rk_random(state);
```

Comment:
Upper bits are zero, `rk_random` is always a 32 bit number stored in a long. Masking won't hurt, but shouldn't be needed.